lib/pure/browsers.nim

```diff
@@ -79,3 +111,36 @@ proc openDefaultBrowser*() {.since: (1, 1).} =
   ## .. code-block:: nim
   ##   block: openDefaultBrowser()
   openDefaultBrowserImpl("http:about:blank")  # See IETF RFC-6694 Section 3.
+
+proc findInstalledBrowsers*(limit = 3.Positive, followSymlinks = true): set[WebBrowsers] {.inline, since: (1, 3).} =
+  ## Try to find *all* installed web browsers, not just the default one.
+  ## * if `followSymlinks` is `true` it will follow symbolic links when searching for web browsers.
+  ## * `limit` is the maximum limit of *installed* web browsers to return.
+  ##
+  ## .. code-block:: nim
+  ##   echo findInstalledBrowsers()  ## {firefox, chromium, safari}
+  var i = 0
+  for x in WebBrowsers:
+    if i > limit: break
+    if findExe($x, followSymlinks).len > 0:
+      inc i
+      incl result, x
+
+proc openBrowsers*(url: string, browsers: openArray[tuple[browser: WebBrowsers, params: string]]) {.since: (1, 3).} =
+  ## Open an `url` with one or more web browsers with arbitrary `params`.
+  ## This does not block. The URL must not be empty string.
+  ## This proc does not raise an exception on error, beware.
```

Comment:
Why wouldn't this raise exceptions for when the browser couldn't be found?

I'm also finding this to be a bit weird to allow opening multiple instances of a web page in different browsers. Wouldn't an `openBrowser` proc that just open exactly one browser works well enough? If users really want to open a page in multiple browsers, they can just do a for-loop over an array, there's no need for fancy things like this.

Actually, wouldn't it be simpler if they just `import osproc` and uses `execShellCmd()`? I don't quite see the point of this proc.

Reply:
The cost of a 1 loop `for` loop is not relevant compared to the time it takes to actually open up the browser with the rendered page,
so basically it works just Ok for 1 url and 1 browser with no options too.

Reply:
I'm not questioning the runtime cost of this proc. I'm questioning whether it's any useful.

This proc can be easily replicated with `execCmdShell()` and a for-loop over the browsers you want to open in, if you ever need such an use case. Not to mention that as with `findInstalledBrowsers`, effectively this proc can only work on Linux.

Reply:
Also, on Linux, from what I understand here, the browsers will be opened one-by-one, and it can only continues after the previously launched browser is closed, making it even less useful.